wai-websockets/Network/Wai/Handler/WebSockets.hs

```diff
@@ -110,4 +111,4 @@ runWebSockets opts req app src sink = do
                 , WS.pendingStream      = stream
                 }
 
-    app pc
+    app pc `finally` WS.close stream
```

Comment:
To avoid async exception issues, it would be better to use `bracket`, something like `bracket (WS.makeStream (...)) WS.close (app . pc)`. Could you make such an update?